IOSExercises/Views/DetailsViewController.m

```diff
@@ -0,0 +1,340 @@
+          @"topLabel" : @"|"
+      },
+      @{
+         @"colorName" : @"Blue",
+        @"buttonName" : @"blueButton",
+          @"selector" : [NSValue valueWithPointer:@selector(chooseColorBlue:)],
+          @"topLabel" : @"[redButton]"
+      },
+      @{
+        @"colorName" : @"Black",
+        @"buttonName" : @"blackButton",
+         @"selector" : [NSValue valueWithPointer:@selector(chooseColorBlack:)],
+         @"topLabel" : @"[blueButton]"
+      }, nil];
+
+    NSMutableDictionary *buttonViews = [[NSMutableDictionary alloc] init];
+
+    for (NSDictionary *color in colors) {
+        UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
+        [button setTranslatesAutoresizingMaskIntoConstraints:NO];
+        [button setTitle:[color objectForKey:@"colorName"]
+                forState:UIControlStateNormal];
+        [button addTarget:self action:[[color objectForKey:@"selector"] pointerValue]
+         forControlEvents:UIControlEventTouchUpInside];
+
+        [controller.view addSubview:button];
+    
+        NSDictionary *views = @{@"button": button};
+        [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:[NSString stringWithFormat:@"H:|[button]|"]
+                                                                                        options:0
+                                                                                        metrics:nil
+                                                                                          views:views]];
+
+        [buttonViews setValue:button
+                       forKey:[color objectForKey:@"buttonName"]];
+
+    }
+
+    [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-20-[redButton]-[blueButton]-[blackButton]"
+                                                                                    options:0
+                                                                                    metrics:nil
+                                                                                      views:buttonViews]];
+
+    [controller setModalPresentationStyle:UIModalPresentationPopover];
+    [self presentViewController:controller
+                       animated:YES
+                     completion:nil];
+
+    UIPopoverPresentationController *popoverController = [controller popoverPresentationController];
+
+    [popoverController setPermittedArrowDirections:UIPopoverArrowDirectionAny];
+
+    [popoverController setSourceView:self.colorLabel];
+    [popoverController setSourceRect:self.colorLabel.bounds];
+}
+
+#pragma mark - Helper Methods
+
+- (void)initDefaultValue {
+    [self.colorLabel setText:@"Black"];
+    [self.birthdayLabel setText:[[self birthdayFormatter] stringFromDate:[NSDate dateWithTimeIntervalSince1970:0]]];
+}
+
+-(NSDateFormatter *)birthdayFormatter {
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    [dateFormatter setDateFormat:@"MMMM d"];
```

Comment:
Make the date format a constant: `static NSString * const kBirthdayDateFormat = @"MMMM d";`

This line comes after the import statements and before the `@interface` declaration for the private category